src/pipert2/core/base/flow.py

```diff
@@ -89,10 +89,17 @@ def execute_event(self, event: Method) -> None:
 
         """
 
-        for routine in self.routines.values():
-            routine.execute_event(event)
-
-        EventExecutorInterface.execute_event(self, event)
+        if event.is_flow_valid(self.name):
+            if event.is_contain_routines(self.name):
+                routines = event.routines_by_flow.get(self.name)
```

Comment:
IMO can be merged to one line and return an empty list if the flow doesn't have routines.

Reply:
But if an event doesn't have any routines that means run ALL of the routines in the specific flow. 
So if it does not contain specific routines -> run all, and if there are routines iterate them.

Reply:
Sorry, i meant that the method `is_contain_routines` can be changed to `event.routines_by_flow.get(self.name)`
and check if the list is empty.

It is technically the same but has less methods in the event.
I personally find it more readable.